src/tools/illink/src/ILLink.RoslynAnalyzer/TrimAnalysis/RequireDynamicallyAccessedMembersAction.cs

```diff
@@ -1,40 +1,55 @@
 // Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
+using System.Reflection.Metadata;
+using Microsoft.CodeAnalysis;
 using ILLink.RoslynAnalyzer.TrimAnalysis;
 using ILLink.Shared.TypeSystemProxy;
+using System.Collections.Immutable;
 
 namespace ILLink.Shared.TrimAnalysis
 {
 	internal partial struct RequireDynamicallyAccessedMembersAction
 	{
+		readonly Compilation _compilation;
```

Comment:
nit: It looks like this `Compilation` is only ever passed down to `ReflectionAccessAnalyzer.TryResolveTypeNameAndMark()` to create the same `TypeNameResolver`. Maybe we could just have a `TypeNameResolver` field instead? And we could then call `TryResolveTypeNameAndMark` directly on that field and remove `ReflectionAccessAnalyzer.TryResolveTypeNameAndMark()`.